vidur/config/config.py

```diff
@@ -1,109 +1,750 @@
+    )
+    skip_cpu_overhead_modeling: bool = field(
+        default=True,
+        metadata={"help": "Whether to skip CPU overhead modeling."},
+    )
+
+
+@dataclass
+class LinearRegressionExecutionTimePredictorConfig(BaseExecutionTimePredictorConfig):
+    polynomial_degree: List[int] = field(
+        default_factory=lambda: list(range(1, 6)),
+        metadata={"help": "Polynomial degree for linear regression."},
+    )
+    polynomial_include_bias: List[bool] = field(
+        default_factory=lambda: [True, False],
+        metadata={"help": "Polynomial include bias for linear regression."},
+    )
+    polynomial_interaction_only: List[bool] = field(
+        default_factory=lambda: [True, False],
+        metadata={"help": "Polynomial interaction only for linear regression."},
+    )
+    fit_intercept: List[bool] = field(
+        default_factory=lambda: [True, False],
+        metadata={"help": "Fit intercept for linear regression."},
+    )
+
+    @staticmethod
+    def get_type():
+        return ExecutionTimePredictorType.LINEAR_REGRESSION
+
+
+@dataclass
+class RandomForrestExecutionTimePredictorConfig(BaseExecutionTimePredictorConfig):
+    num_estimators: List[int] = field(
+        default_factory=lambda: [250, 500, 750],
+        metadata={"help": "Number of estimators for random forest."},
+    )
+    max_depth: List[int] = field(
+        default_factory=lambda: [8, 16, 32],
+        metadata={"help": "Maximum depth for random forest."},
+    )
+    min_samples_split: List[int] = field(
+        default_factory=lambda: [2, 5, 10],
+        metadata={"help": "Minimum samples split for random forest."},
+    )
+
+    @staticmethod
+    def get_type():
+        return ExecutionTimePredictorType.RANDOM_FORREST
+
+
+@dataclass
+class ClusterConfig:
+    num_replicas: int = field(
+        default=1,
+        metadata={"help": "Number of replicas."},
+    )
+    replica_config: ReplicaConfig = field(default_factory=ReplicaConfig)
+    execution_time_predictor_config: BaseExecutionTimePredictorConfig = field(
+        default_factory=RandomForrestExecutionTimePredictorConfig,
+        metadata={"help": "Execution time predictor config."},
+    )
+    global_scheduler_config: BaseGlobalSchedulerConfig = field(
+        default_factory=RoundRobinGlobalSchedulerConfig,
+        metadata={"help": "Global scheduler config."},
+    )
+    replica_scheduler_config: BaseReplicaSchedulerConfig = field(
+        default_factory=SarathiSchedulerConfig,
+        metadata={"help": "Replica scheduler config."},
+    )
+    metrics_config: MetricsConfig = field(
+        default_factory=MetricsConfig,
+        metadata={"help": "Metrics config."},
+    )
+
+    def update_predictor_config(self):
+        if "{DEVICE}" in self.execution_time_predictor_config.compute_input_file:
+            self.execution_time_predictor_config.compute_input_file = self.execution_time_predictor_config.compute_input_file.replace(
```

Comment:
this should move to execution_time_predictor code? also we can reduce the code bloat by doing a general find and replace in all the input file attributes